main.go

```diff
@@ -120,6 +126,19 @@ func main() {
 		os.Exit(1)
 	}
 
+	directClient, err := client.New(clientConf,
+		client.Options{Scheme: mgr.GetScheme(), Mapper: mgr.GetRESTMapper()})
+	if err != nil {
+		setupLog.Error(err, "failed to create direct client")
+		os.Exit(1)
+	}
+
+	// run migration logic before controllers start
+	if err := migrate.Migrate(stopCtx, setupLog.WithName("migrate"), directClient); err != nil {
```

Comment:
Question: do we want some flag to disable migration ? for an un-foreseen case where we would not want to run it ?

Reply:
ok, I added "disable-migration" flag to optional turn off migration logic

Reply:
@e0ne Are you talking about the flag or migration logic in general? 
If about the flag, then think about it in a broader context. For me, this flag is a way to completely disable migration logic. Later, the migration handler can contain much more logic than it has now, and it is better to have a way to disable it.

Reply:
I prefer to keep as few config options as possible. This PR adds some flag with we don't need at the moment and also it requires to re-build image or create a custom helm chart to enable it

Reply:
Ok, let's remove the flag. The current logic is relatively safe, and there should be no reason to disable it now.

Reply:
My fear that later on as migration logics are added, we will forget to add this flag.
then we might encounter cases where we wish we had it.

Reply:
After discussion we decided to add env variable which will turn off all migrations